### PR TITLE
Adobe Analytics

### PR DESCRIPTION
**Pull request by ruchiramoitra**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/250)

### CODE REVIEW

File: integrations/AdobeAnalytics/browser.js [hunk 1]
```diff
@@ -0,0 +1,1159 @@
...
+      return merchMap;
+    }
+
+    each((rudderProp) => {
+      // if property mapped with products. as starting handle differently
+      if (rudderProp.productMerchProperties.startsWith("products.")) {
+        const key = rudderProp.productMerchProperties.split(".");
+        // take the keys after products. and find the value in properties
+        const value = _.get(properties, key[1]);
+        if (value && value !== "undefined") {
```
Comment (arnab-p) [inline integrations/AdobeAnalytics/browser.js:1014]: Check similar implementations and figure out if we can move it to a util method

File: integrations/integration_cname.js [hunk 2]
```diff
@@ -70,6 +70,9 @@ const commonNames = {
   PINTEREST_TAG: "PINTEREST_TAG",
   pinterest: "PINTEREST_TAG",
   Pinterest: "PINTEREST_TAG",
+  "Adobe Analytics": "ADOBE_ANALYITCS",
+  ADOBE_ANALYTICS: "ADOBE_ANALYTICS",
+  AdobeAnalytics: "ADOBE_ANALYTICS",
```
[the remark this answers is not in the document]
Comment (ruchiramoitra) [reply integrations/integration_cname.js:75]: done

File: integrations/AdobeAnalytics/util.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-param-reassign */
+import _ from "lodash";
+
+/**
+ * @param  {} value
+ * @param  {} return true or false accordingly if value is empty
+ */
+
+function isEmpty(value) {
```
[the remark this answers is not in the document]
Comment (ruchiramoitra) [reply integrations/AdobeAnalytics/util.js]: done

File: integrations/AdobeAnalytics/util.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  return (
+    value === undefined ||
+    value === null ||
+    (typeof value === "object" && Object.keys(value).length === 0) ||
+    (typeof value === "string" && value.trim().length === 0)
```
[the remark this answers is not in the document]
Comment (ruchiramoitra) [reply integrations/AdobeAnalytics/util.js]: done

File: integrations/AdobeAnalytics/util.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-param-reassign */
```
[the remark this answers is not in the document]
Comment (ruchiramoitra) [reply integrations/AdobeAnalytics/util.js:1]: done

File: integrations/AdobeAnalytics/browser.js [hunk 6]
```diff
@@ -0,0 +1,1159 @@
...
+      bitrate || 0,
+      startupTime || 0,
+      fps || 0,
+      droppedFrames || 0
+    );
+  }
+
+  // Begin heartbeat implementation
+
+  populatHeartbeat(rudderElement) {
```
[the remark this answers is not in the document]
Comment (ruchiramoitra) [reply integrations/AdobeAnalytics/browser.js]: done

File: integrations/AdobeAnalytics/browser.js [hunk 7]
```diff
@@ -0,0 +1,1159 @@
...
+  updateCommonWindowSKeys(rudderElement) {
+    const { properties, type, context } = rudderElement.message;
+    let campaign;
+    if (context && context.campaign) {
+      campaign = context.campaign.name;
+    } else {
+      campaign = properties.campaign;
+    }
+    const channel = rudderElement.message.channel || properties.channel;
+    const { state, zip } = properties;
```
[the remark this answers is not in the document]
Comment (ruchiramoitra) [reply integrations/AdobeAnalytics/browser.js]: done

**Review by arnab-p — APPROVED** [review]

**Pull request merged**